.github/workflows/publish.yml

```diff
@@ -68,17 +68,14 @@ jobs:
           path: |
             ~/.npm
             node_modules
-          key: npm-v16-${{ runner.os }}-refs/heads/master-${{ hashFiles('package.json') }}
+          key: npm-v16-${{ runner.os }}-refs/heads/v2-${{ hashFiles('package.json') }}
 
       - name: Install Node.js and npm
         uses: actions/setup-node@v1
         with:
           node-version: 16.x
           registry-url: https://registry.npmjs.org
 
-      # Normally we have a guarantee that deps are already there, still it may not be the case when:
-      # - `master` build for same commit failed (and we still pushed tag manually)
-      # - We've pushed tag manually before `master` build finalized
       - name: Install dependencies
         if: steps.cacheNpm.outputs.cache-hit != 'true'
```

Comment:
I've just spotted that we miss `id: cacheNpm` on `Retrieve dependencies from cache` step (without that this condition is ineffective)